extensions/mapstore/src/main/java/com/hazelcast/mapstore/Queries.java

```diff
@@ -43,36 +45,121 @@ class Queries {
-                queryParams(columnMetadata.size()));
+    private static String buildLoadAllKeysQuery(String mapping, String idColumn) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT ");
+        DIALECT.quoteIdentifier(sb, idColumn);
+        sb.append(" FROM ");
+        DIALECT.quoteIdentifier(sb, mapping);
+        return sb.toString();
+    }
 
-        String setClause = columnMetadata.stream()
-                .filter(cm -> !idColumn.equals(cm.getName()))
-                .map(cm -> '\"' + cm.getName() + "\" = ?")
-                .collect(joining(", "));
+    private String buildStoreSinkQuery(String mapping, List<SqlColumnMetadata> columnMetadata) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SINK INTO ");
+        DIALECT.quoteIdentifier(sb, mapping);
+        sb.append(" (");
+        for (Iterator<SqlColumnMetadata> iterator = columnMetadata.iterator(); iterator.hasNext(); ) {
+            SqlColumnMetadata column = iterator.next();
+            DIALECT.quoteIdentifier(sb, column.getName());
+            if (iterator.hasNext()) {
+                sb.append(", ");
+            }
+        }
+        sb.append(") VALUES (");
+        appendQueryParams(sb, columnMetadata.size());
+        sb.append(')');
+        return sb.toString();
+    }
 
-        storeUpdate = String.format("UPDATE \"%s\" SET %s WHERE \"%s\" = ?", mapping, setClause, idColumn);
+    private String buildStoreUpdateQuery(String mapping, String idColumn, List<SqlColumnMetadata> columnMetadata) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("UPDATE ");
+        DIALECT.quoteIdentifier(sb, mapping);
+        sb.append(" SET ");
+        for (Iterator<SqlColumnMetadata> iterator = columnMetadata.iterator(); iterator.hasNext(); ) {
+            SqlColumnMetadata column = iterator.next();
+            if (idColumn.equals(column.getName())) {
+                continue;
+            }
+            DIALECT.quoteIdentifier(sb, column.getName());
+            sb.append(" = ?");
+            if (iterator.hasNext()) {
+                sb.append(", ");
+            }
+        }
+        sb.append(" WHERE ");
+        DIALECT.quoteIdentifier(sb, idColumn);
+        sb.append(" = ?");
+        return sb.toString();
+    }
 
-        delete = String.format("DELETE FROM \"%s\" WHERE \"%s\" = ?", mapping, idColumn);
+    private static String buildDeleteQuery(String mapping, String idColumn) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("DELETE FROM ");
+        DIALECT.quoteIdentifier(sb, mapping);
+        sb.append(" WHERE ");
+        DIALECT.quoteIdentifier(sb, idColumn);
+        sb.append(" = ?");
+        return sb.toString();
+    }
 
-        deleteAllFactory = n -> String.format("DELETE FROM \"%s\" WHERE \"%s\" IN (%s)", mapping, idColumn,
-                queryParams(n));
+    private static void appendQueryParams(StringBuilder sb, int n) {
```

Comment:
nit : I think this is easier to read
```
private static void appendQueryParams(StringBuilder sb, int n) {
        IntStream.rangeClosed(1, n)
                .forEach(i -> {
                    sb.append('?');
                    if (i < n) {
                        sb.append(", ");
                    }
                });
    }
```

Reply:
or:
IntStream.rangeClosed(1, n).mapToObj(i -> "?").collect(joining(","))